pkg/console/status/status.go

```diff
@@ -102,26 +102,66 @@ func setConditionValue(conditionType string, err error) operatorsv1.ConditionSta
 	return operatorsv1.ConditionFalse
 }
 
-func IsDegraded(operatorConfig *operatorsv1.Authentication) bool {
-	for _, condition := range operatorConfig.Status.Conditions {
-		if strings.HasSuffix(condition.Type, operatorsv1.OperatorStatusTypeDegraded) &&
-			condition.Status == operatorsv1.ConditionTrue {
-			return true
+type StatusHandler struct {
+	client      v1helpers.OperatorClient
+	statusFuncs []v1helpers.UpdateStatusFunc
+}
+
+func (c *StatusHandler) AddCondition(newStatusFunc v1helpers.UpdateStatusFunc) {
+	c.statusFuncs = append(c.statusFuncs, newStatusFunc)
+}
+
+func (c *StatusHandler) AddConditions(newStatusFuncs []v1helpers.UpdateStatusFunc) {
+	for _, newStatusFunc := range newStatusFuncs {
+		c.statusFuncs = append(c.statusFuncs, newStatusFunc)
+	}
+}
+
+func (c *StatusHandler) FlushAndReturn(returnErr error) error {
+	if _, _, updateErr := v1helpers.UpdateStatus(c.client, c.statusFuncs...); updateErr != nil {
+		return updateErr
+	}
+	return returnErr
+}
+
+func (c *StatusHandler) UpdateObservedGeneration(newObservedGeneration int64) {
+	generationFunc := func(oldStatus *operatorsv1.OperatorStatus) error {
+		oldStatus.ObservedGeneration = newObservedGeneration
+		return nil
+	}
+	c.statusFuncs = append(c.statusFuncs, generationFunc)
+}
+
+func (c *StatusHandler) UpdateReadyReplicas(newReadyReplicas int32) {
+	generationFunc := func(oldStatus *operatorsv1.OperatorStatus) error {
+		oldStatus.ReadyReplicas = newReadyReplicas
+		return nil
+	}
+	c.statusFuncs = append(c.statusFuncs, generationFunc)
+}
+
+func (c *StatusHandler) UpdateDeploymentsLastGeneration(newLastGeneration int64) {
+	generationFunc := func(oldStatus *operatorsv1.OperatorStatus) error {
+		generationFound := false
```

Comment:
Please use library-go's `SetDeploymentGeneration` instead of reinventing it:

https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourcemerge/apps.go#L48

Here's an example of use in the openshift-apiserver operator:

https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/workload/workload_openshiftapiserver_v311_00_sync.go#L175

This also relates to something I mentioned on slack to you - the the provision of the generation argument to `ApplyDeployment` should be updated to use library-go's `ExpectedDeploymentGeneration`:

https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourcemerge/apps.go#L40

Example of use in openshift-apiserver operator: 

https://github.com/openshift/cluster-openshift-apiserver-operator/blob/master/pkg/operator/workload/workload_openshiftapiserver_v311_00_sync.go#L367

vs what the console operator currently does (retrieves the generation from the target deployment rather than the operator status):

https://github.com/openshift/console-operator/blob/master/pkg/console/operator/sync_v400.go#L225